overlord/state/change.go

```diff
@@ -311,6 +311,22 @@ func (c *Change) Status() Status {
 	return c.status
 }
 
+// Wait return true if any task is in "Wait" state.
+//
+// XXX: This should probably be part of Status() but the issue with
+// Status() is that "Doing" has higher priority than "Wait" so refresh
+// tasks that have a "check-rerefresh" task will always be in doing
+// state even when "link-snap" is in wait state.
```

Comment:
I fear we might need a mechanism to mark check-rerefresh to be treated specially in Status. Its status should only be taken into account once the rest of the tasks in the change are ready. I think that should mostly do what we need?